xs/src/libslic3r/Polyline.hpp

```diff
@@ -36,15 +36,26 @@ class Polyline : public MultiPoint {
     Polygons grow(double delta, double scale = CLIPPER_OFFSET_SCALE, ClipperLib::JoinType joinType = ClipperLib::jtSquare, double miterLimit = 3.0) const;
 };
 
+
+/// ThickPolyline : a polyline with a width for each point
+/// This calss has a vector of coordf_t, it must be the same size than points.
+/// it's used to store the size of the line at this point.
+/// Also, the endpoint let us know if the front() and back() of the polyline 
+/// join something or is a dead-end.
 class ThickPolyline : public Polyline {
     public:
+    /// width size must be == point size
     std::vector<coordf_t> width;
+    /// if true => it's an endpoint, if false it join an other ThickPolyline. first is at front(), second is at back()
     std::pair<bool,bool> endpoints;
```

Comment:
(From performance perspective, it may be worth to work with group of ThickPolylines, remembering enpoint connectivity)

Reply:
I don't think it's worth it.
Premature optimisation is not a good practice. If the medial_axis is a hog now, i can devot some time to see with where and why.

Reply:
Yes, it was just a remark.
Lot of code is looping to find incident lines, it may be even a bit easier to understand